src/disco/shred/test_shredder.c

```diff
@@ -141,6 +141,109 @@ test_shredder_pcap( void ) {
 #endif /* FD_HAS_HOSTED */
 
 
+static void
+test_multiple_shredders( void ) {
+  fd_rng_t _rng[ 1 ]; fd_rng_t * r = fd_rng_join( fd_rng_new( _rng, 0U, 0UL ) );
+
+  signer_ctx_t signer_ctx[ 1 ];
+  signer_ctx_init( signer_ctx, test_private_key );
+
+  #define SHREDDERS 4
+
+  FD_TEST( SHREDDERS>0 );
+
+  fd_shredder_t   _shredders[ SHREDDERS ];
+  fd_shredder_t *  shredders[ SHREDDERS ];
+  /* Initialize all the shredders */
+  for( ulong i=0; i<SHREDDERS; i++ ) {
+    FD_TEST( &_shredders[ i ]==fd_shredder_new( &_shredders[ i ], test_signer, signer_ctx, (ushort)0 ) );
+    shredders[ i ] = fd_shredder_join( &_shredders[ i ] );
+    FD_TEST( shredders[ i ] );
+  }
+
+  fd_entry_batch_meta_t meta[ 1 ];
+  fd_memset( meta, 0, sizeof( fd_entry_batch_meta_t ) );
+
+  uchar data_shreds  [ 2048UL*FD_REEDSOL_DATA_SHREDS_MAX   ] = { 0 };
+  uchar parity_shreds[ 2048UL*FD_REEDSOL_PARITY_SHREDS_MAX ] = { 0 };
+
+  fd_fec_set_t _set[ 1 ];
+  for( ulong j=0UL; j<FD_REEDSOL_DATA_SHREDS_MAX;   j++ ) _set->data_shreds[   j ] = data_shreds + 2048UL*j;
+  for( ulong j=0UL; j<FD_REEDSOL_PARITY_SHREDS_MAX; j++ ) _set->parity_shreds[ j ] = parity_shreds + 2048UL*j;
+
+  ulong data_shred_cnt   = 0;
+  ulong parity_shred_cnt = 0;
+  ulong idx  = 0UL;
+  ulong slot = 1UL;
+
+  ulong data_sz = 1024UL*1024UL;
+  uchar data    [ 1024UL*1024UL ];
+  for( ulong i=0; i<data_sz; i++ ) data[ i ] = fd_rng_uchar( r );
+
+  /* Randomly choose a shredder to process a batch until the data buffer is exhausted. */
+  while( idx<data_sz ) {
+    ulong sz        = ( fd_rng_ulong_roll( r, 4UL )+1UL )*25000UL;
```

Comment:
How about just making this a random number between 1 and 100k? No need for it to be a multiple of 25k.